logstash-core/lib/logstash/logging/logger.rb

```diff
@@ -79,5 +80,35 @@ def self.get_logging_context
         return @@logging_context
       end
     end
+
+    class SlowLogger
+      def initialize(name, warn_threshold, info_threshold, debug_threshold, trace_threshold)
+        slowlog_name = ["slowlog", name].join('.')
+        @slowlogger = LogManager.getLogger(slowlog_name)
+        @warn_threshold = warn_threshold
+        @info_threshold = info_threshold
+        @debug_threshold = debug_threshold
+        @trace_threshold = trace_threshold
+      end
+
+      def as_data(event, time_in_nanos)
```

Comment:
I think we need more context here. At least the filter ID and filter config. Imagine if there were 8 grok filters and you'd have to find the slowest. With the existing code, you'll know grok is slow, but which one?